pkg/kubelet/volume/empty_dir/empty_dir.go

```diff
@@ -197,36 +201,51 @@ func (ed *emptyDir) GetPath() string {
 
 // TearDown simply discards everything in the directory.
 func (ed *emptyDir) TearDown() error {
-	switch ed.medium {
-	case api.StorageTypeDefault:
-		return ed.teardownDefault()
-	case api.StorageTypeMemory:
-		return ed.teardownTmpfs()
-	default:
-		return fmt.Errorf("unknown storage medium %q", ed.medium)
+	return ed.TearDownAt(ed.GetPath())
+}
+
+// TearDownAt simply discards everything in the directory.
+func (ed *emptyDir) TearDownAt(dir string) error {
+	// Figure out the medium.
+	if medium, err := ed.mediumer.GetMedium(dir); err != nil {
+		glog.Errorf("TIM: error from GetMedium: %v", err)
+		return err
+	} else {
+		switch medium {
+		case mediumMemory:
+			ed.medium = api.StorageTypeMemory
```

Comment:
hurrrr, I am guess this is running against all-tmpfs storage, tricking this
code.  But if so, all the tmpfs code should be failing the same way

On Mon, Mar 16, 2015 at 2:31 PM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/kubelet/volume/empty_dir/empty_dir.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5167#discussion_r26529629>
> :
>
> > -		return ed.teardownTmpfs()
> > -	default:
> > -		return fmt.Errorf("unknown storage medium %q", ed.medium)
> > +	return ed.TearDownAt(ed.GetPath())
> > +}
> > +
> > +// TearDownAt simply discards everything in the directory.
> > +func (ed *emptyDir) TearDownAt(dir string) error {
> > +	// Figure out the medium.
> > +	if medium, err := ed.mediumer.GetMedium(dir); err != nil {
> > +		glog.Errorf("TIM: error from GetMedium: %v", err)
> > +		return err
> > +	} else {
> > +		switch medium {
> > +		case mediumMemory:
> > +			ed.medium = api.StorageTypeMemory
>
> @thockin <https://github.com/thockin> This is where the tests are failing
> on Linux but passing on OS X.
>
> "medium == 1" here because of this line:
> https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/volume/empty_dir/empty_dir_linux.go#L35
>
> SetUp goes through ed.setupDefault() while TearDown goes through
> ed.teardownTempfs()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5167/files#r26529629>
> .
>